### PR TITLE
PYIC-3803 Put exit code audit change behind feature flag.

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Put audit event exit code change behind feature flag.

### What changed
Change made to include extension field exitCode only when feature flag (exitCodes) is enabled.

### Issue tracking

- [PYIC-3803](https://govukverify.atlassian.net/browse/PYIC-3803)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-3803]: https://govukverify.atlassian.net/browse/PYIC-3803?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java [hunk 1]
```diff
@@ -195,7 +196,7 @@ private void sendIdentityIssuedAuditEvent(
                         ipvSessionItem.getVot(),
                         ipvSessionItem.isCiFail(),
                         contraIndicators.hasMitigations(),
-                        userIdentity.getExitCode());
+                        configService.enabled(EXIT_CODES) ? userIdentity.getExitCode() : null);
```
Comment (Joe-Edwards-GDS) [inline lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java:199]: I'm not sure this is totally necessary - the exit code should already be `null` if the flag is not enabled. I suppose it adds an extra layer of certainty.

**Review by Joe-Edwards-GDS — DISMISSED** [review]

File: lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java (hunk 1 above)
Comment (rpayal) [reply lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java:199]: Ticket says - Returning Failure Codes in the IPV_IDENTITY_ISSUED audit event is Controlled with a Feature Toggle

If not needed then will close this PR.

Comment (rpayal) [reply lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java:199]: As part of 3947 feature flag is removed. Will close this and then update the ticket accordingly.

**Comment by rpayal** [comment]
Closing as already handled in user identity service as part of other ticket.

**Pull request closed without merging**